src/Services/EncounterService.php

```diff
@@ -242,6 +242,54 @@ public function getEncounterForPatient($pid, $eid)
         return sqlQuery($sql, array($pid, $eid));
     }
 
+    public function insertEncounter($pid, $data)
+    {
+
+
+        $sql .= " INSERT INTO form_encounter SET";
+        $sql .= "     date =NOW(),";
+        $sql .= "     onset_date = ?,";
+        $sql .= "     reason = ?,";
+        $sql .= "     facility = ?,";
+        $sql .= "     pc_catid = ?,";
+        $sql .= "     facility_id = ?,";
+        $sql .= "     billing_facility = ?,";
+        $sql .= "     sensitivity = ?,";
+        $sql .= "     referral_source = ?,";
+        $sql .= "     pid = ?,";
+        $sql .= "     encounter = ?,";
+        $sql .= "     pos_code = ?,";
+        $sql .= "     external_id = ?,";
+        $sql .= "     provider_id = ?,";
+        $sql .= "     parent_encounter_id = ?";
+        
+        $results = sqlInsert(
+            $sql,
+            [
+                $data["onset_date"],
+                $data["reason"],
+                $data["facility"],
+                $data["pc_catid"],
+                $data["facility_id"],
+                $data["billing_facility"],
+                $data["sensitivity"],
+                $data["referral_source"],
+                $data["pid"],
+                $data["encounter"],
```

Comment:
note will create `encounter` via the generate_id() function

Reply:
Okay